### ISSUE TITLE
"This competition is ongoing" the day before the competition

### ISSUE THREAD

**Issue by SqAtx** [thewca/worldcubeassociation.org#1716]
I just noticed a small thing when looking at Vancouver Open Summer 2017 competition page.

I get a message saying "Note: This competition is ongoing. Come back after June 10, 2017 to see the results!".

As it is currently 10:05 pm PDT (UTC-8) on June 9th, 2017, I did not expect to see that message :)

**Comment by jfly** [comment]
Yeah, our `in_progress` method uses a half baked heuristic. See https://github.com/thewca/worldcubeassociation.org/blob/58097ba2fb751058371cc1a9d627650c0054b901/WcaOnRails/app/models/competition.rb#L639-L641.

The fundamental problem here is that we do not know the schedules of competitions, nor do we know their timezones. We're slowly working on fixing that, but it's a lot work with very little gain.

Some relevant code here: https://github.com/thewca/worldcubeassociation.org/blob/58097ba2fb751058371cc1a9d627650c0054b901/WcaOnRails/app/models/competition.rb#L643-L649, and a relevant discussion here: https://github.com/thewca/worldcubeassociation.org/pull/1206/files#r98485399.

**Comment by AlbertoPdRF** [comment]
Now that we do know the schedules and timezones of competitions, we could change those methods, no? I'm labeling this as a good second issue, although I'd like your confirmation @jfly.

**Comment by jfly** [comment]
Yep! It should be possible to do a better job now. The schedule doesn't always end up matching reality, but I can't immediately think of a situation where that would cause any serious issues.
